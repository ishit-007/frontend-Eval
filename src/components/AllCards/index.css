.cards-container {
    justify-content: space-around;
  display: flex;
  flex-wrap: wrap;
  gap: 10%;
  width: 60%;
  margin-left: 20%;
}
