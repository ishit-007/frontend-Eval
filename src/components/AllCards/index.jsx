/* eslint-disable no-unused-vars */
import './index.css';
import React from 'react';
import Card from '../Card';
import Search from '../Search';

import { makeRequest, BACKEND_URL, GET_EVENTS_DATA } from '../../util/makeRequest';

const AllCards = () => {
  const [searchedText, setSearchedText] = React.useState('');
  const [matchingEvents, setMatchingEvents] = React.useState([]);
  const [events, setEvents] = React.useState([]);
  const sortEventsByDate = events => {
    events.sort((a, b) => {
      const date1 = new Date(a.datetime);
      const date2 = new Date(b.datetime);
      return date1 - date2;
    });
  };
  React.useEffect(() => {
    makeRequest(GET_EVENTS_DATA, {}).then(data => {
      sortEventsByDate(data);
      setEvents(data);
    });
  }, []);
  return (
    <>
      <div className="all-Cards">
        <Search searchedText={searchedText} setSearchedText={setSearchedText} />
      </div>
      <div className="cards-container">
        {events.map(event => {
          if (!event.name.toLowerCase().includes(searchedText.toLowerCase())) {
            return null;
          }
          return <Card props={event} key={event.id} />;
        })}
      </div>
    </>
  );
};

export default AllCards;
