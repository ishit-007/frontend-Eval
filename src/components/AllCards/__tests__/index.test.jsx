import React from 'react';
import AllCards from '..';
import renderer from 'react-test-renderer';
describe('AllCards', () => {
  it('should render the AllCards component', async () => {
    const tree = renderer.create(<AllCards />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
