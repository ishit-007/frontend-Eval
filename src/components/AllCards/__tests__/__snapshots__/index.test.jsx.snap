// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllCards should render the AllCards component 1`] = `
Array [
  <div
    className="all-Cards"
  >
    <div
      className="search"
    >
      <input
        onChange={[Function]}
        placeholder="EVENT NAME"
        style={
          Object {
            "borderColor": undefined,
          }
        }
        type="text"
      />
      <i
        className="fa fa-search"
        style={
          Object {
            "fil": undefined,
          }
        }
      />
    </div>
    <div
      className="checkbox-filter-container"
    >
      <input
        id="all"
        name="all"
        onChange={[Function]}
        type="checkbox"
        value="all"
      />
      <label>
        All
      </label>
      <br />
      <input
        id="register"
        name="register"
        onChange={[Function]}
        type="checkbox"
        value="register"
      />
      <label>
        Registered
      </label>
      <br />
      <input
        id="Bookmark"
        name="Bookmarked"
        onChange={[Function]}
        type="checkbox"
        value="Bookmarked"
      />
      <label>
        Bookmarked
      </label>
      <br />
      <input
        id="available"
        name="available"
        onChange={[Function]}
        type="checkbox"
        value="available"
      />
      <label>
        Seats Available
      </label>
    </div>
  </div>,
  <div
    className="cards-container"
  />,
]
`;
